tests/test_updater_ng.py

```diff
@@ -326,6 +333,69 @@ def test_non_existing_target_file(self) -> None:
         with self.assertRaises(exceptions.DownloadHTTPError):
             self.updater.download_target(info)
 
+    # TAP 14 tests ~ REMOVE COMMENT BEFORE THE NEXT COMMIT!!
+
+    # test case to check for the TAP 14 folder
+    def test_check_folder_tap14(self) -> None:
+        # Creating the parent folder for the TAP 14 folder
+        self.assertTrue(os.path.isdir(self.tap14_directory))
+
+    def test_check_tap14_contents(self) -> None:
+        # Checking specific files inside TAP 14
+        filenames = ["targets.json", "root.json"]
+        for file in filenames:
+            self.assertTrue(
+                os.path.isfile(os.path.join(self.tap14_directory, file))
+            )
+
+        # Checking specific folders inside TAP 14
+        foldernames = ["targets", "1.0.0"]
+        for folder in foldernames:
+            self.assertTrue(
+                os.path.isdir(os.path.join(self.tap14_directory, folder))
+            )
+
+    def test_get_spec_version1(self) -> None:
+        # This uses the default SUPPORTED_VERSIONS variable from updater.py
+        with self.assertRaises(exceptions.DownloadError):
+            _get_spec_version(
+                ["1", "2", "3"], "4", ngclient.updater.SUPPORTED_VERSIONS
+            )
+
+        self.assertEqual(
+            _get_spec_version(
+                ["1", "2", "3"], "3", ngclient.updater.SUPPORTED_VERSIONS
+            ),
+            ("3", None),
+        )
+
+    def test_get_spec_version2(self) -> None:
+        warningchecker = "Not using the latest specification version available on the repository"
+        # Checks with different values
```

Comment:
When you have this many test cases, I recommend putting them in a list:

```python
test_cases = [
    (["1", "2", "3"], "3", ["3", "5", "6"],  "3", False),
    (["3", "5", "6"], "3", ["1", "2", "3", "4"], "3", True),
]
for repo_versions, spec_version, supported_versions, expected_version, should_have_warning in test_cases:
    actual_version, warning = _get_spec_version(...)
    self.assertEqual(actual_version, expected_version)
    self.assertEqual(bool(warning), should_have_warning)
# Do something similar for error_test_cases
```

Reply:
Then you can have a comment for each test case explaining why you have it.

Reply:
So, should I add a `msg` or leave a comment? I guess doing both would be repetitive.

Reply:
Yeah, if you're going to put a `msg` in the asserts you don't need a comment.

Reply:
My bad actually. I'll have to put comments if I'll be using the for loop.